lib/cocoapods/installer/analyzer.rb

```diff
@@ -238,18 +238,20 @@ def update_repositories
       #
       def copy_embedded_target_pod_targets_to_host(aggregate_target, embedded_aggregate_targets)
         return if aggregate_target.requires_host_target?
-        # Get the uuids of the aggregate_target's user_targets' embedded targets if any
-        embedded_uuids = Set.new(aggregate_target.user_targets.map do |target|
-          aggregate_target.user_project.embedded_targets_in_native_target(target).map(&:uuid)
-        end.flatten)
-        return if embedded_uuids.empty?
-        embedded_aggregate_targets.each do |embedded_target|
-          next unless embedded_target.user_targets.map(&:uuid).any? do |embedded_uuid|
-            embedded_uuids.include? embedded_uuid
+        pod_target_names = Set.new aggregate_target.pod_targets.map(&:name)
+        aggregate_user_target_uuids = Set.new aggregate_target.user_targets.map(&:uuid)
+        embedded_aggregate_targets.each do |embedded_aggregate_target|
+          next unless embedded_aggregate_target.user_targets.any? do |embedded_user_target|
+            # You have to ask the host target's project for the host targets of
+            # the embedded target, as opposed to asking user_project for the
+            # embedded targets of the host target. The latter doesn't work when
+            # the embedded target lives in a sub-project. The line below gets
+            # the host target uuids for the embedded target and checks to see if
+            # those match to any of the user_target uuids in the aggregate_target.
+            !aggregate_user_target_uuids.intersection(Set.new(aggregate_target.user_project.host_targets_for_embedded_target(embedded_user_target).map(&:uuid))).empty?
```

Comment:
I originally didn't because I thought rubocop would complain about unneeded variable assignment (or something to that effect), but it didn't! This is much better.